cmd/defaults.go

```diff
@@ -0,0 +1,98 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/containers/storage"
+	"k8s.io/klog/v2"
+)
+
+// Return "chroot" if we know we're not actually root, "oci" otherwise.
+func builderDefaultIsolation() (string, error) {
+	if inUserNamespace() {
+		// We probably don't have enough privileges to use a proper runtime.
+		return "chroot", nil
```

Comment:
If we are in a user namespace, does this make us vulnerable to https://access.redhat.com/security/cve/CVE-2021-20182?

Reply:
In an unprivileged container, the builder container will already be have a device control group configured for limiting which devices the kernel will allow direct access to, even when the nodes exist, and `/dev` will have the smaller set of entries in it which we grant to unprivileged containers.  I expect we'll also be running without CAP_MKNOD.

Reply:
fwiw @adambkaplan we also dove into the CVE ramifications back in March with https://github.com/openshift/builder/pull/220#discussion_r590595420

bottom line:  "we appear good"

that said, I think @nalind 's comment here ^^ would make a good comment when we see this code again in 6 months and ask the same question, not remembering this discussion :-)

Reply:
Adding `Lean on the container that we're in being itself unprivileged (i.e., having control groups including the
device cgroup configured for us, being provided with a smaller set of devices in /dev, and likely running without a
few capabilities that we don't need), and reduce the degree of isolation that we try to use to what we know we're
actually allowed to do in an unprivileged container.`

Reply:
thanks @nalind